staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -87,6 +93,17 @@ type authProxyRoundTripper struct {
 	rt http.RoundTripper
 }
 
+// NewCacheRoundTripper creates a roundtripper that reads the ETag on
+// response headers and send the If-None-Match header on subsequent
+// corresponding requests.
+func NewCacheRoundTripper(cacheDir string, rt http.RoundTripper) http.RoundTripper {
+	d := diskv.New(diskv.Options{BasePath: cacheDir})
+	t := httpcache.NewTransport(diskcache.NewWithDiskv(d))
```

Comment:
Hrm. I still don't quite understand. The spec seems to talk specifically about [how caching layers should propagate](https://tools.ietf.org/html/rfc7234#section-4.3.2) 304 when `If-Modified-Since` is used:

```
If an If-None-Match header field is not present, a request containing
an If-Modified-Since header field (Section 3.3 of [RFC7232])
indicates that the client wants to validate one or more of its own
stored responses by modification date.  A cache recipient SHOULD
generate a 304 (Not Modified) response (using the metadata of the
selected stored response) if one of the following cases is true: 1)
the selected stored response has a Last-Modified field-value that is
earlier than or equal to the conditional timestamp; 2) no
Last-Modified field is present in the selected stored response, but
it has a Date field-value that is earlier than or equal to the
conditional timestamp; or, 3) neither Last-Modified nor Date is
present in the selected stored response, but the cache recorded it as
having been received at a time earlier than or equal to the
conditional timestamp.
```

Based on that, I'm not sure why I as a client (via the caching layer) wouldn't receive a 304 in this case. I must be misunderstanding. Maybe this is a discussion for the httpcache project or a mailing list instead? I don't want to bog you all down with my questions if the caching layer semantics are ultimately a third-party concern with this change.